test/OpenTelemetry.Tests/Implementation/Trace/ActivitySourceAdapterTest.cs

```diff
@@ -0,0 +1,186 @@
+﻿// <copyright file="ActivitySourceAdapterTest.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using OpenTelemetry.Tests.Implementation.Testing.Export;
+using OpenTelemetry.Trace;
+using Xunit;
+
+namespace OpenTelemetry.Tests.Implementation.Trace
+{
+    public class ActivitySourceAdapterTest : IDisposable
+    {
+        private TestSampler testSampler;
+        private TestActivityProcessor testProcessor;
+        private ActivitySourceAdapter activitySourceAdapter;
+
+        static ActivitySourceAdapterTest()
+        {
+            Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+            Activity.ForceDefaultIdFormat = true;
+        }
+
+        public ActivitySourceAdapterTest()
+        {
+            this.testSampler = new TestSampler();
+            this.testProcessor = new TestActivityProcessor();
+            this.activitySourceAdapter = new ActivitySourceAdapter(this.testSampler, this.testProcessor);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ActivitySourceAdapterCallsStartStopActivityProcessor(bool isSampled)
+        {
+            this.testSampler.SamplingAction = (samplingParameters) =>
+            {
+                return new SamplingResult(isSampled);
+            };
+
+            bool startCalled = false;
+            bool endCalled = false;
+            this.testProcessor.StartAction =
+                (a) =>
+                {
+                    startCalled = true;
+                    if (isSampled)
```

Comment:
nit: per code the else of this `if` statement is never reached, right? Perhaps assert directly that the recorded flag must be on if the code reaches here.